akka-sample-sharding-scala/killrweather/src/main/scala/sample/killrweather/Aggregator.scala

```diff
@@ -1,125 +0,0 @@
- * aggregate, averages, high/low, topK (e.g. the top N highest temperatures).
- */
-private[killrweather] trait Aggregator {
-
-  import Aggregator._
-  import WeatherRoutes.{QueryResult, TimeWindow}
-
-  def TypeKey: EntityTypeKey[Aggregator.Command]
-
-  protected def aggregating(entityId: String, values: Vector[Data]): Behavior[Aggregator.Command] =
-    Behaviors.setup { context =>
-      Behaviors.receiveMessage {
-        case Aggregate(data, received, replyTo) =>
-          val updated = values :+ data
-          context.log.debug(
-            s"${updated.size} total readings from station ${entityId}, " +
-              s"average ${average(updated.map(_.temperature))}, diff: processingTime - eventTime: ${received - data.eventTime}ms")
-
-          replyTo ! WeatherRoutes.DataIngested(data.wsid)
-
-          aggregating(entityId, updated) // store
-
-        case Query(wsid, dataType, func, replyTo) =>
-          // typically you would separate read/write, here for simplicity we show both
-          val value = dataType match {
-            case DataType.Temperature =>
-              func match {
-                case Function.Average =>
-                  val start: Long = values.headOption.map(_.eventTime).getOrElse(0)
-                  val end: Long = values.lastOption.map(_.eventTime).getOrElse(0)
-                  Vector(TimeWindow(start, end, average(values.map(_.temperature))))
-
-                case Function.HighLow =>
-                  val (start, min) = values.map(e => e.eventTime -> e.temperature).min
-                  val (end, max) = values.map(e => e.eventTime -> e.temperature).max
-                  Vector(TimeWindow(start, end, min), TimeWindow(start, end, max))
-
-                case Function.Current =>
-                  Vector(values.lastOption
-                    .map(e => TimeWindow(e.eventTime, e.eventTime, e.temperature))
-                    .getOrElse(TimeWindow(0, 0, 0.0)))
-              }
-
-            case _ => ???
-          }
-
-          replyTo ! QueryResult(wsid, dataType.tag, func.tag, values.size, value)
-
-          Behaviors.same
-      }
-    }
-}
-
-/**
- * Generally geo-based data, unless it is a floating station.
- *
- * This is just an example: cluster sharding would be overkill for just keeping a small amount of data,
- * but becomes useful when you have a collection of 'heavy' actors (in terms of processing or state)
- * so you need to distribute them across several nodes.
- */
-private[killrweather] object WeatherAggregator extends Aggregator {
-
-  val TypeKey: EntityTypeKey[Aggregator.Command] =
-    EntityTypeKey[Aggregator.Command]("WeatherStation")
-
-  def init(system: ActorSystem[_]): Unit =
-    ClusterSharding(system).init(Entity(TypeKey) { entityContext =>
-      WeatherAggregator(entityContext.entityId)
-    })
-
-  def apply(id: String): Behavior[Aggregator.Command] = aggregating(id, Vector.empty)
-
-}
```

Comment:
Mentioned above, renamed this to WeatherStation since it only ever was used for data for a single station.